File: 0001b-spreadsheets/solution.py
from typing import List


def solution():
    case_count = int(input())

    for _ in range(case_count):
        text = input()

        # R23C55 -> BC23
        pos = text.find("C")
        if text[0] == "R" and text[1].isdigit() and 1 < pos < len(text):
            row = int(text[1:pos])
            col = int(text[pos + 1:])

            text_list: List[str] = []
            while col > 0:
                if col % 26 == 0:
                    text_list.append("Z")
                    col = (col - 1) // 26
                else:
                    text_list.append(chr(ord("A") + col % 26 - 1))
                    col //= 26

            for c in reversed(text_list):
                print(c, end="")
            print(row)

        # BC23 -> R23C55
        else:
            for pos, c in enumerate(text):
                if c.isdigit():
                    break

            row, col = text[:pos], text[pos:]

            col_sum = 0
            for c in row:
                col_sum = col_sum * 26 + ord(c) - ord("A") + 1

            print(f"R{col}C{col_sum}")


if __name__ == "__main__":
    solution()

File: 0001b-spreadsheets/solution.cpp
#include <iostream>
#include <string>
#include <vector>
#include <cstdlib>


void solution() {
    int case_count;
    std::cin >> case_count;

    std::string text;
    for (int i = 0; i < case_count; ++i) {
        std::cin >> text;

        // R23C55 -> BC23
        long pos = text.find('C');
        if ('R' == text[ 0 ] && isdigit(text[1]) && 1 < pos && pos < text.size()) {
            long row = atol(text.substr(1, pos - 1).c_str());
            long col = atol(text.substr(pos + 1).c_str());

            std::vector<char> new_text;
            while (col > 0) {
                if (col % 26 == 0) {
                    new_text.push_back('Z');
                    col = (col - 1) / 26;
                } else {
                    new_text.push_back('A' + (col % 26 - 1));
                    col /= 26;
                }
            }

            while (!new_text.empty()) {
                std::cout << *new_text.rbegin();
                new_text.pop_back();
            }

            std::cout << row << '\n';

        // BC23 -> R23C55
        } else {
            long pos = -1;
            for (pos = 0; pos < text.size(); ++pos) {
                if (isdigit(text[pos])) {
                    break;
                }
            }

            std::string row = text.substr(0, pos);
            std::string col = text.substr(pos);

            long c = 0;
            for (std::string::const_iterator p = row.begin(); p != row.end(); ++p) {
                c = c * 26 + (*p - 'A' + 1);
            }

            std::cout << 'R' << col << 'C' << c << '\n';
        }
    }
}


void setup() {
    std::ios_base::sync_with_stdio(false);
    std::cin.tie(NULL);
}


int main() {
    setup();
    solution();
}
